packages/js/src/Modules/Verto/webrtc/helpers.ts

```diff
@@ -563,14 +593,59 @@ function getWebRTCSupportedBrowserList(): Array<IWebRTCSupportedBrowser> {
           features: ['video', 'audio'],
           supported: SUPPORTED_WEBRTC.full,
         },
-        { browserName: 'Firefox', features: ['audio'], supported: SUPPORTED_WEBRTC.partial },
+        {
+          browserName: 'Firefox',
+          features: ['audio'],
+          supported: SUPPORTED_WEBRTC.partial,
+        },
         { browserName: 'Safari', supported: SUPPORTED_WEBRTC.not_supported },
-        { browserName: 'Edge', features: ['audio'], supported: SUPPORTED_WEBRTC.partial },
+        {
+          browserName: 'Edge',
+          features: ['audio'],
+          supported: SUPPORTED_WEBRTC.partial,
+        },
       ],
     },
   ];
 }
 
+function createAudio(file, id): HTMLAudioElement | null {
+  if (file && id) {
+    const ringAudio = document.createElement('audio');
+    ringAudio.id = id;
+    ringAudio.loop = true;
+    ringAudio.src = file;
+    ringAudio.load();
+    document.body.appendChild(ringAudio);
+    return ringAudio;
+  }
+  return null;
+}
+
+function playAudio(audioElement: HTMLAudioElement): void {
+  if (audioElement) {
+    audioElement
+      .play()
+      .then(() => {
+        // when resolved we can call pause();
+      })
+      .catch((error) => {
+        console.error('playAudio', error);
+      });
+  }
+}
+
+function stopAudio(audioElement: HTMLAudioElement): void {
+  // Add delay to wait until play() returns the promise
+  // https://developers.google.com/web/updates/2017/06/play-request-was-interrupted
+  setTimeout(() => {
+    if (audioElement) {
+      audioElement.pause();
+      audioElement.currentTime = 0;
+    }
+  }, 1000);
+}
```

Comment:
Setting a timeout could be really hard to debug, especially since the 1 second seems fairly arbitrary--if the ringtone audio file is large, it may take longer to load, or if its small, it could load much quicker.

I would store the promise somehow, maybe like this (untested):
```suggestion
function playAudio(audioElement: HTMLAudioElement): void {
  if (audioElement) {
    audioElement._playFulfilled = false;
    audioElement._promise = audioElement
      .play();
      
    audioElement._promise.then(() => {
        audioElement._playFulfilled = true;
      })
      .catch((error) => {
        console.error('playAudio', error);
        
        audioElement._playFulfilled = true;
      });
  }
}

function stopAudio(audioElement: HTMLAudioElement): void {
  if (!audioElement) return;
  // Add delay to wait until play() returns the promise
  // https://developers.google.com/web/updates/2017/06/play-request-was-interrupted
  if (audioElement._playFulfilled) {
      audioElement.pause();
      audioElement.currentTime = 0;
  } else if (audioElement._promise && audioElement._promise.then) {
    audioElement._promise.then(() => {
      audioElement.pause();
      audioElement.currentTime = 0;
    })
  } else {
    setTimeout(() => {
      if (audioElement) {
        audioElement.pause();
        audioElement.currentTime = 0;
      }
    }, 1000);
  }
}
```

Reply:
Done! It worked very well.